service/api_logic/sports_logic.py

```diff
@@ -19,22 +19,80 @@ def get_all_sports(session):
 
 
 @handle_exceptions
-def get_all_leagues_by_sport(session, dto: SportsLeagueDTO):
-    offset = (dto.page - 1) * dto.per_page
-    leagues = (
-        session.query(League)
-        .join(Sport, Sport.sport_id == League.sport_id)
-        .filter(Sport.sport_name == dto.sport)
-        .limit(dto.per_page)
-        .offset(offset)
-        .all()
+def get_all_leagues_by_sport(session, filters_dto: SportsLeagueDTO):
+    query = (
+        session.query(
+            League.sport_id,
+            League.league_id,
+            League.logo,
+            League.name,
+            League.country,
+            League.api_id
+        )
+        .join(Sport, League.sport_id == Sport.sport_id)
     )
+
+    model_aliases = {
+        "leagues": League,
+    }
+
+    query = apply_filters(query, filters_dto.to_dict(), model_aliases)
+
+    offset, limit = filters_dto.get_pagination()
+    if offset is not None and limit is not None:
+        query = query.offset(offset).limit(limit)
+
+    leagues = query.all()
+    count = query.count()
```

Comment:
WE still have to move it to DAL, but, why would we do another DB call, when we can just count in Python, or in JavaScript